Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1164,40 +1164,40 @@ private static int GetNumMajorCantrips(int tier)
             if (tier < 4)
                 return numMajors;
 
-            int majorCantripChance = ThreadSafeRandom.Next(1, 100);
+            int majorCantripChance = ThreadSafeRandom.Next(1, 10000);
 
             switch (tier)
             {
                 case 4:
                     // 10% chance for a major, also
-                    if (majorCantripChance > 90)
+                    if (majorCantripChance < 1000)
                         numMajors = 1;
                     break;
                 case 5:
                     // 20% chance for a major
-                    if (majorCantripChance > 80)
+                    if (majorCantripChance < 2000)
                         numMajors = 1;
                     break;
                 case 6:
-                    // 20% chance for a major, 5% chance for 2 majors
-                    if (majorCantripChance > 80)
+                    // 20% chance for a major, 1% chance for 2 majors
+                    if (majorCantripChance < 2000)
                         numMajors = 1;
-                    else if (majorCantripChance > 95)
+                    else if (majorCantripChance < 100)
                         numMajors = 2;
                     break;
                 case 7:
                 default:
-                    // 30% chance for a major, 20% chance for 2 majors,
-                    // 10% chance for 3 majors, 5% chance for 4 majors, and 2% chance for 5 majors
-                    if (majorCantripChance > 70)
+                    // 30% chance for a major, 10% chance for 2 majors,
+                    // 1% chance for 3 majors, 0.1% chance for 4 majors, and 0.01% chance for 5 majors
+                    if (majorCantripChance < 3000)
                         numMajors = 1;
-                    else if (majorCantripChance > 80)
+                    else if (majorCantripChance < 1000)
                         numMajors = 2;
-                    else if (majorCantripChance > 90)
+                    else if (majorCantripChance > 100)
                         numMajors = 3;
-                    else if (majorCantripChance > 95)
+                    else if (majorCantripChance > 10)
                         numMajors = 4;
-                    else if (majorCantripChance > 98)
+                    else if (majorCantripChance > 1)
```

Comment:
>= 1? not sure.

Just thinking about what happens when the random is the min or max values in the range provided.

Reply:
Forgot to flip the > symbol to <